EchoPro/survey.py

```diff
@@ -36,741 +23,440 @@ class Survey:
-
-        Parameters
-        ----------
-        selected_transects : list or None
-            The subset of transects used in the calculations
-        """
-
-        self.bio_calc = None
-        self.bio_calc = ComputeTransectVariables(self)
-        self.bio_calc.get_transect_results_gdf(selected_transects)
-
-        # create Dataset containing useful distributions and variables over length and age
-        self.bio_calc.bin_ds = generate_bin_ds(self)
-
-        # add NASC_adult to transect_results_gdf (needs to occur after generate_bin_ds)
-        self.bio_calc.set_adult_NASC()
-
-        # calculate Jolly-Hampton CV
-        self.bio_calc.jollyhampton_cv = self.run_cv_analysis(kriged_data=False)
-
-        # compute abundance and biomass for each length and age bin
-        self.compute_length_age_variables(data="transect")
+        Loads the biological, NASC, and stratification
+        data using parameters obtained from the configuration
+        files. This will generate data attributes associated with the tags
+        defined in both the configuration yml files and the reference CONFIG_MAP
+        and LAYER_NAME_MAP dictionaries.
+        """
+
+        ### Check whether data files defined from the configuration file exists
+        # Generate flat JSON table comprising all configuration parameter names
+        flat_configuration_table = pd.json_normalize(self.config).filter(regex="filename")
+
+        # Parse the flattened configuration table to identify data file names and paths
+        parsed_filenames = flat_configuration_table.values.flatten()
+
+        # Append the root directory to the parsed data file names
+        data_paths = [Path(self.config['data_root_dir'] + '/' + file) for file in parsed_filenames]
+
+        # Evaluate whether either file is missing
+        data_existence = [path.exists() for path in data_paths]
+
+        # Assign the existence status to each configuration file for error evaluation
+        # Error evaluation and print message (if applicable)
+        if not all(data_existence):
+            missing_data = [ files for files, exists in zip( parsed_filenames, data_existence ) if not exists ]
+            raise FileNotFoundError(f"The following data files do not exist: {missing_data}")
+        
+        # Initialize data attributes ! 
+        setattr( self , 'acoustics' , { } )
+        setattr( self , 'biology' , { } )
+        setattr( self , 'spatial' , { } )
+        setattr( self , 'statistics' , { } )
+
+        ### Data validation and import
+        # Iterate through each file defined/detected in 'data_paths' and 'parsed_filenames'
+
+        for file , path in zip(parsed_filenames, data_paths):
+
+            # Parse config to find appropriate settings
+            config_name = flat_configuration_table.apply(lambda row: row[row == file].index, axis=1)
+            
+            # Convert to string
+            config_identifier = config_name[0].tolist()[0].lower()
```

Comment:
I don't think you need to use apply for simple row selection? I have not tried to evaluate this, because there's path issue when I try to run the code, but I suspect if you use simple selection you would not have to do hard code `[0]` type of selection.

Reply:
For reference, I am trying to retrieve the column name that matches the parsed_filename (e.g. `'Biological/US/2019_biodata_length.xlsx'`). So when running at least these two lines, they yield:
`config_name = flat_configuration_table.apply(lambda row: row[row == file].index, axis=1)`
---> `Index(['biological.length.US.filename'], dtype: object`

`config_identifier = config_name[0].tolist()[0].lower()`
---> `'biological.length.us.filename'`

When I use simple selection, this is certainly possible such that I select the columns directly:
`flat_configuration_table.columns[flat_configuration_table.isin([file]).any()]` ---> `Index(['biological.length.US.filename'], dtype='object')`

But this then requires the same hard-coding to convert from this to `'biological.length.us.filename'` (i.e. `config_identifier`), albeit in simpler fashion:
`config_name = flat_configuration_table.columns[flat_configuration_table.isin([file]).any()]`
`config_name[0].lower()` ---> `'biological.length.us.filename'`

This may become moot w.r.t suggestions related to `L173-179` (i.e. `for data_type in ["biological", "stratification", "NASC", "kriging"]: `).